### PR TITLE
Ref: replace sentry/typescript by sentry-internal/typescript

### LINKED ISSUE #3379: Replace @sentry/typescript by @sentry-internal/typescript
### Description

The devDependency uses [@sentry/typescript](https://www.npmjs.com/package/@sentry/typescript) where the last publish date was from 3 years ago

We should instead use [@sentry-internal/typescript](https://www.npmjs.com/package/@sentry-internal/typescript) that has the same version number as the other JavaScript packages.

The change of names from the package happened here: https://github.com/getsentry/sentry-javascript/pull/2808

### PR DESCRIPTION
**Pull request by lucas-zimerman**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [X] Refactoring

## :scroll: Description

Sentry/Typescript was deprecated by 3 years and was replaced by sentry-internal/typescript.
And by the addition of Sentry-internal/Typescript, new errors appeared, mostly in regard to not exporting certain Sentry parameters as type.

Lastly, this shouldn't impact the end users so I didn't update the changelog.

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [X] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [X] All tests passing
- [ ] No breaking changes

## :crystal_ball: Next steps

#skip-changelog

Close #3379

### CODE REVIEW

File: scripts/update-javascript.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@ set -euo pipefail
 
 tagPrefix=''
 repo="https://github.com/getsentry/sentry-javascript.git"
-packages=('@sentry/browser' '@sentry/core' '@sentry/hub' '@sentry/integrations' '@sentry/react' '@sentry/types' '@sentry/utils')
+packages=('@sentry/browser' '@sentry/core' '@sentry/hub' '@sentry/integrations' '@sentry/react' '@sentry-internal/types' '@sentry/utils')
```
Comment (krystofwoldrich) [inline scripts/update-javascript.sh:6]:
```suggestion
packages=('@sentry/browser' '@sentry/core' '@sentry/hub' '@sentry/integrations' '@sentry/react' '@sentry/types' '@sentry/utils' '@sentry-internal/typescript')
```

I think this was a typo, @sentry/types exists.

**Comment by krystofwoldrich** [comment]
Looks good. Thank you. I have only one comment to the updater script and then we can merge this.

File: scripts/update-javascript.sh (hunk 1 above)
Comment (lucas-zimerman) [reply scripts/update-javascript.sh]: true, sorry for that, ll fix it

**Review by krystofwoldrich — APPROVED** [review]
Thank you 🚀 Looks good.

**Pull request merged**